### PR TITLE
feat(chat): add code app logs (Issue #2484)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

Need to add ability to view logs while code app is running

Issues:

- Issue #2484 

**UI changes**

Снимок экрана 2024-11-01 в 23 58 31
Снимок экрана 2024-11-02 в 00 01 32

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Derikyan** [comment]
/deploy-review
>- **Deployment failed**: [canceled](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/533725)

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2503.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                    <div key={index} className="flex flex-col gap-2">
+                      <LogLines logContent={log.content} />
+                    </div>
+                  ))}
+                </React.Fragment>
+              );
+            })
+          ) : (
+            <p>No logs found</p>
+          )}
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: maybe lets move this to LogsView component?

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                  {selectedApplicationLogs?.logs?.map((log, index) => (
+                    <div key={index} className="flex flex-col gap-2">
+                      <LogLines logContent={log.content} />
+                    </div>
+                  ))}
+                </React.Fragment>
+              );
+            })
+          ) : (
+            <p>No logs found</p>
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: translation

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    <Modal
+      portalId="chat"
+      state={isOpen ? ModalState.OPENED : ModalState.CLOSED}
+      dataQa="marketplace-application-logs"
+      overlayClassName="!z-40"
+      containerClassName="flex w-full flex-col min-h-[350px] xl:max-w-[820px] max-w-[800px]"
+      onClose={onClose}
+    >
+      <div className="px-3 pb-4 pt-6 md:px-6">
+        <h2 className="text-base font-semibold">Application logs</h2>
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: translation

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      </div>
+      {selectedLogsLoading ? (
+        <div className="flex w-full grow items-center justify-center rounded-t  p-4">
+          <Spinner size={30} className="mx-auto" />
+        </div>
+      ) : (
+        <div className="flex grow flex-col items-center justify-center gap-4 overflow-y-auto break-all px-3 pb-6 md:px-6">
+          {selectedApplicationLogs?.logs.length ? (
+            selectedApplicationLogs?.logs.map((log) => {
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]:
```suggestion
            selectedApplicationLogs.logs.map((log) => {
```

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+              return (
+                <React.Fragment key={log.content}>
+                  {selectedApplicationLogs?.logs?.map((log, index) => (
+                    <div key={index} className="flex flex-col gap-2">
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: locContent isn't uniq value?

File: apps/chat/src/store/application/application.reducers.ts [hunk 6]
```diff
@@ -117,6 +122,21 @@ export const applicationSlice = createSlice({
...
+    getLogsSuccess: (
+      state,
+      { payload }: PayloadAction<ApplicationLogsType>,
+    ) => {
+      state.logsLoading = false;
+      state.appLogs = payload;
+    },
+    getLogsFail: (state) => {
+      state.logsLoading = false;
+      state.error = true;
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/application/application.reducers.ts]: maybe we can replace this with logsLoadingStatus? and use UploadStatus enum?

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx (hunk 5 above)
Comment (Derikyan) [reply apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: i think yes

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      {selectLogsLoading ? (
+        <div className="flex w-full grow items-center justify-center rounded-t  p-4">
+          <Spinner size={30} className="mx-auto" />
+        </div>
+      ) : (
+        <div className="flex grow flex-col items-center justify-center gap-4 overflow-y-auto break-all px-3 pb-6 md:px-6">
+          {selectApplicationLogs?.logs.length ? (
+            <LogsView logs={selectApplicationLogs.logs} />
+          ) : (
+            <p>{t('No logs found')}</p>
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]: please, move all this logic to LogsView (74-77 lines)

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2503.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/components/Marketplace/ApplicationLogs.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    .split('\n')
+    .map((line, index) => <p key={index}>{line.replace(ansiRegex, '')}</p>);
+};
+
+interface LogsViewProps {
+  logs: { content: string; instance: string }[];
+}
+
+const LogsView = ({ logs }: LogsViewProps) => (
+  <React.Fragment>
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/ApplicationLogs.tsx]:
```suggestion
  <>
```

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**